src/components/structures/TimelinePanel.js

```diff
@@ -270,6 +333,9 @@ var TimelinePanel = React.createClass({
 
             debuglog("TimelinePanel: paginate complete backwards:"+backwards+"; success:"+r);
 
+            // Unpaginate events gratuitously after every paginate
```

Comment:
I feel like this might fit more naturally as a callback from ScrollPanel (`onUnfillRequest` as a counterpart to `onFillRequest`, I suppose). Just a thought though; YMMV.